internal/pkg/agent/application/upgrade/step_mark.go

```diff
@@ -277,11 +278,50 @@ func CleanMarker(log *logger.Logger, dataDirPath string) error {
 // LoadMarker loads the update marker. If the file does not exist it returns nil
 // and no error.
 func LoadMarker(dataDirPath string) (*UpdateMarker, error) {
-	return loadMarker(markerFilePath(dataDirPath))
+	return loadMarker(markerFilePath(dataDirPath), true)
 }
 
-func loadMarker(markerFile string) (*UpdateMarker, error) {
-	markerBytes, err := readMarkerFile(markerFile)
+func UpdateMarkerFile(markerFile string, opts ...func(*UpdateMarker)) error {
+	fileLock, err := newMarkerFileLocker(markerFile)
+	if err != nil {
+		return fmt.Errorf("creating update marker locker for reading: %w", err)
+	}
+	err = fileLock.Lock()
+	if err != nil {
+		return fmt.Errorf("locking update marker file %q for reading: %w", markerFile, err)
+	}
+	defer func(fileLock Locker) {
+		errUnlock := fileLock.Unlock()
+		if errUnlock != nil {
+			err = goerrors.Join(err, fmt.Errorf("unlocking marker file after reading: %w", errUnlock))
+		}
+	}(fileLock)
+
+	marker, err := loadMarker(markerFile, false)
+	if err != nil {
+		return fmt.Errorf("failed to read marker file during updating: %w", err)
+	}
+
+	for _, opt := range opts {
+		opt(marker)
+	}
+
+	if err := saveMarkerToPath(marker, markerFile, true, false); err != nil {
+		return fmt.Errorf("failed updating marker file: %w", err)
+	}
+
+	return nil
+}
+
+func loadMarker(markerFile string, isSafe bool) (*UpdateMarker, error) {
```

Comment:
Is there a specific reason why somebody would want to load the update marker with an isSafe=false ? I am not seeing the usecase here..